### PR TITLE
feat: intergrate theme for single Button component

### PR DESCRIPTION
**Pull request by loc-nguyenthien**
## Summary
- Adding default theme for Button component
- Adding other Button style: ButtonPrimary, ButtonSecondary, ButtonOutline, ButtonTransparent
- Allow default theming for these customize Button

## Check List
- [ ] Your code builds clean without any errors or warnings.
- [ ] Check coding convention.
- [ ] Updated Storybook components and stories to reflect any changes made to the UI.
- [ ] Test covers all edge cases and coverage is greater than or equal to 80%.

## Proof of Completeness
Screenshot 2023-11-06 at 15 26 11
Screenshot 2023-11-07 at 11 18 30

### CODE REVIEW

File: src/theme/components/index.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import {
+  ButtonTheme,
+  ButtonOutlineTheme,
+  ButtonPrimaryTheme,
+  ButtonSecondaryTheme,
+  ButtonTransparentTheme,
+} from './Button'
+
+export default {
+  Button: ButtonTheme,
```
Comment (chungnguyenhoang) [inline src/theme/components/index.ts:10]: don't need to required

Comment (loc-nguyenthien) [reply src/theme/components/index.ts:10]: This is the default theme for Button (if we don't config in the project `extendTheme`), so I don't know what you mean by not required

File: src/components/Button/Button.tsx [hunk 2]
```diff
@@ -44,21 +52,28 @@ const Button: React.FC<ButtonProps> = ({
...
-    {...props}>
-    <Label color={textColor}>{text}</Label>
-  </ButtonWrapper>
-)
+}) => {
+  const ButtonTheme = useTheme().components.Button
+  return (
+    <ButtonWrapper
+      onPress={onPress}
+      activeOpacity={0.8}
```
Comment (hangnguyensaigontech) [inline src/components/Button/Button.tsx:62]: should we need to include value activeOpacity in theme as well?

File: src/components/Button/Button.tsx [hunk 3]
```diff
@@ -72,7 +87,7 @@ const ButtonWrapper = styled.TouchableOpacity(
   }: Omit<ButtonProps, 'text' | 'onPress'> & {theme?: ITheme}) => ({
     paddingVertical: theme?.spacing.small,
     paddingHorizontal: theme?.spacing.slim,
-    borderRadius: borderRadius || metrics.borderRadius,
+    borderRadius: borderRadius,
```
Comment (hangnguyensaigontech) [inline src/components/Button/Button.tsx:90]:
```suggestion
    borderRadius,
```

File: src/components/Button/Button.tsx [hunk 4]
```diff
@@ -83,7 +98,7 @@ const ButtonWrapper = styled.TouchableOpacity(
 )
 
 const Label = styled.Text(({theme, color}: {color?: string; theme?: ITheme}) => ({
-  color: color || 'white',
+  color: color,
```
Comment (hangnguyensaigontech) [inline src/components/Button/Button.tsx:101]: same with borderRadius
```suggestion
  color,
```

File: src/components/Button/ButtonOutline.tsx [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+  textColor,
+  backgroundColor,
+  outlineColor,
+  outlineWidth,
+  disabled,
+  ...props
+}) => {
+  const ButtonOutlineTheme = useTheme().components.ButtonOutline
+  return (
+    <Button
```
Comment (honghoangsts) [inline src/components/Button/ButtonOutline.tsx:16]: `<Button
      outline
      {...ButtonOutlineTheme}
      {...props}
    />
`

File: src/components/Button/ButtonPrimary.tsx [hunk 6]
```diff
@@ -0,0 +1,19 @@
+import React from 'react'
+import {useTheme} from '../../hooks'
+import type {ButtonProps} from './Button'
+import Button from './Button'
+
+const ButtonPrimary: React.FC<ButtonProps> = ({textColor, backgroundColor, disabled, ...props}) => {
+  const ButtonPrimaryTheme = useTheme().components.ButtonPrimary
+  return (
+    <Button
```
Comment (honghoangsts) [inline src/components/Button/ButtonPrimary.tsx:9]: <Button outline {... ButtonPrimaryTheme} {...props} />

File: src/components/Button/ButtonSecondary.tsx [hunk 7]
```diff
@@ -0,0 +1,21 @@
+import React from 'react'
+import {useTheme} from '../../hooks'
+import type {ButtonProps} from './Button'
+import Button from './Button'
+
+const ButtonSecondary: React.FC<ButtonProps> = ({textColor, backgroundColor, disabled, ...props}) => {
+  const ButtonSecondaryTheme = useTheme().components.ButtonSecondary
+  return (
+    <Button
```
Comment (honghoangsts) [inline src/components/Button/ButtonSecondary.tsx:9]: <Button outline {... ButtonSecondaryTheme} {...props} />

File: src/components/Button/ButtonTransparent.tsx [hunk 8]
```diff
@@ -0,0 +1,21 @@
+import React from 'react'
+import {useTheme} from '../../hooks'
+import type {ButtonProps} from './Button'
+import Button from './Button'
+
+const ButtonTransparent: React.FC<ButtonProps> = ({textColor, backgroundColor, disabled, ...props}) => {
+  const ButtonTransparentTheme = useTheme().components.ButtonTransparent
+  return (
+    <Button
```
Comment (honghoangsts) [inline src/components/Button/ButtonTransparent.tsx:9]: <Button outline {... ButtonTransparentTheme} {...props} />

**Review by honghoangsts — CHANGES_REQUESTED** [review]

File: src/components/Button/Button.tsx (hunk 2 above)
Comment (loc-nguyenthien) [reply src/components/Button/Button.tsx:62]: We actually don't need to, the props is extends from `TouchableOpacityProperties` so it already have `activeOpacity`. And since we pass all the props into with `{...props}`, if `activeOpacity` exist, it will override the above value

File: src/components/Button/Button.tsx (hunk 3 above)
Comment (loc-nguyenthien) [reply src/components/Button/Button.tsx:90]: Updated

File: src/components/Button/Button.tsx (hunk 4 above)
Comment (loc-nguyenthien) [reply src/components/Button/Button.tsx:101]: Updated

File: src/components/Button/ButtonOutline.tsx (hunk 5 above)
Comment (loc-nguyenthien) [reply src/components/Button/ButtonOutline.tsx:16]: Updated

File: src/components/Button/ButtonPrimary.tsx (hunk 6 above)
Comment (loc-nguyenthien) [reply src/components/Button/ButtonPrimary.tsx]: Updated

File: src/components/Button/ButtonSecondary.tsx (hunk 7 above)
Comment (loc-nguyenthien) [reply src/components/Button/ButtonSecondary.tsx]: Updated

File: src/components/Button/ButtonTransparent.tsx (hunk 8 above)
Comment (loc-nguyenthien) [reply src/components/Button/ButtonTransparent.tsx]: Updated

**Review by hangnguyensaigontech — APPROVED** [review]

**Review by honghoangsts — APPROVED** [review]
